### PR TITLE
docs(annex): add informative Conformance Test Corpus (T-16)

### PR DESCRIPTION
**Pull request by aorzelskiGH**
## Summary

Add a new informative annex `Conformance Test Corpus` to IDTA-01004 with a structured, technology-neutral set of acceptance tests that implementations of the Access Rule Model can run to demonstrate conformance.

## Problem

IDTA-01004 currently ships example rules but no structured catalogue of test / acceptance criteria. Reviewers and implementers rediscover the same edge cases on every release: default-deny behaviour, permit-overrides combination, treatment of inapplicable rules (`$aasdesc` in Repository profiles), runtime formula errors, filter combination semantics, RIGHTS mapping for PUT CREATE/UPDATE, supplementalSemanticIds scoping, etc.

## Solution

- Introduce `documentation/IDTA-01004/modules/ROOT/pages/annex/conformance-test-corpus.adoc`.
- Organize tests into five groups: Rule loading, Rule evaluation, RIGHTS / operation mapping, Supplemental semantic IDs, Descriptor applicability per profile.
- Each test has a stable ID (`RULE.load.001`, `EVAL.combine.001`, `RIGHT.map.003`, `SUPP.rule.001`, `DESC.reg.001`, ...), a one-line purpose, an expected behaviour and a requirement level.
- Register the annex in `nav.adoc`.
- Aligned with a matching corpus in IDTA-01002.

## Affected files

- `documentation/IDTA-01004/modules/ROOT/pages/annex/conformance-test-corpus.adoc` (new)
- `documentation/IDTA-01004/modules/ROOT/nav.adoc`

## Review notes

- The corpus is explicitly marked *informative*. No normative surface area is changed; the MUST/SHOULD columns refer back to normative requirements defined in the Access Rule Model and the new Evaluation Semantics section (T-15).
- Companion PR in `aas-specs-api` adds a matching corpus for the Query Language and HTTP/REST API.
- Intent: a follow-up work stream can turn each row into an executable test case.

Refs: Review Finding T-16

### CODE REVIEW

File: documentation/IDTA-01004/modules/ROOT/pages/annex/conformance-test-corpus.adoc [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+| If no rule matches a request, the default decision MUST be `deny`.
+| Status `403` (or transport equivalent).
+
+| EVAL.combine.001
+| If any matching rule grants access, the decision MUST be `permit` (permit-overrides).
+| Status `200` (or transport equivalent).
+
+| EVAL.error.001
+| A rule that is inapplicable (e.g. references `$aasdesc` on a non-Registry) MUST be treated as "not applicable" (no match, no error).
+| Does not contribute to the decision.
```
Comment (Martin187187) [inline documentation/IDTA-01004/modules/ROOT/pages/annex/conformance-test-corpus.adoc:68]: in my opinion the comparison that contains a non aplicabble data should be false

**Comment by Martin187187** [comment]
opinion from TF: dont add new requirements in conformance test. Many of those "tests" make sense and could be added to the standard directly?

**Comment by Martin187187** [comment]
This change has too many issues and would make conformance too strict. It introduces new specification requirements that are not defined in the standard.

Before adding strict conformance tests for this behavior, the logic should first be added to the standard itself. These tests could also be useful as examples in the annex, showing how the security layer should handle specific cases.
